cloudevents-sdk-paho-mqtt/src/mqtt_producer_record.rs

```diff
@@ -0,0 +1,150 @@
+use super::headers;
+use crate::headers::MqttVersion::MQTT_5;
+use cloudevents::event::SpecVersion;
+use cloudevents::message::{
+    BinaryDeserializer, BinarySerializer, Error, MessageAttributeValue, Result,
+    StructuredDeserializer, StructuredSerializer,
+};
+use cloudevents::Event;
+use paho_mqtt::{MessageBuilder, Properties, Property, PropertyCode};
+use std::option::Option::Some;
+
+pub struct MessageRecord {
+    pub(crate) headers: Properties,
+    pub(crate) payload: Option<Vec<u8>>,
+}
+
+impl MessageRecord {
+    /// Create a new empty [`MessageRecord`]
+    pub fn new() -> Self {
+        MessageRecord {
+            headers: Properties::new(),
+            payload: None,
+        }
+    }
+
+    pub fn from_event(event: Event, version: &headers::MqttVersion) -> Result<Self> {
+        match version {
+            headers::MqttVersion::MQTT_5 => {
+                BinaryDeserializer::deserialize_binary(event, MessageRecord::new())
+            }
```

Comment:
As I understood MQTTv5 should support Binary and Structured Content Mode.
https://github.com/cloudevents/spec/blob/v1.0.1/mqtt-protocol-binding.md#32-structured-content-mode

So I would add a parameter to define the content mode (Binary or Structured) and not the MQTT version directly.

Reply:
More than `version` you would use the `Encoding` struct? sounds good